snf-cyclades-app/synnefo/userdata/models.py

```diff
@@ -57,6 +57,17 @@ class PublicKeyPair(ProfileModel):
     content = models.TextField(validators=[
         MaxLengthValidator(SSH_KEY_MAX_CONTENT_LENGTH)])
     fingerprint = models.CharField(max_length=100, null=False, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    # NOTE(KostasPelelis): This fields should always be null and
+    #                      false respectively
+    deleted_at = models.DateTimeField(null=True, default=None)
+    deleted = models.BooleanField(null=False, default=False)
+    # NOTE(KostasPelelis): Type field should take values between ssh and x509
+    #                      Django does not currently support
+    #                      SQL ENUM type so we wiil just use
+    #                      a plain CharField
+    type = models.CharField(max_length=10, null=False, default='ssh')
```

Comment:
Add a `unique_together` constraint for the (user, name) tuple. This serves both as a database constraint to ensure database integrity and as an index to allow for faster searches. (Your frequent queries should be indexed for performance reasons).